Changes to build against PG13. This includes the removal of a referen…

jim-mlodgenski opened the pull request:
…ce to OPAQUEOID which has been deprecated long before PLV8 was invented.

jim-mlodgenski committed:
Changes to build against PG13. This includes the removal of a reference to OPAQUEOID which has been removed.

jim-mlodgenski committed:
Changes to build against PG13. This includes the removal of a reference to OPAQUEOID which has been deprecated long before PLV8 was invented.

JerrySievert commented:
when did it become deprecated?  from conversations while plv8 was being written in 2011, it appears that it was still a valid way to check for a trigger.

is there a version cutoff where this behavior changed?

jim-mlodgenski commented:
On Fri, Jun 26, 2020 at 10:51 AM Jerry Sievert <notifications@github.com>
wrote:

> when did it become deprecated? from conversations while plv8 was being
> written in 2011, it appears that it was still a valid way to check for a
> trigger.
>
It was there to support functions created back in 7.3 and before.

> is there a version cutoff where this behavior changed?
>
Anything created after 8.2 would never have OPAQUEOID as a return type
https://www.postgresql.org/message-id/4110.1583255415%40sss.pgh.pa.us

—
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plv8/plv8/pull/397#issuecomment-650221414>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJFFZ2BX6GTHWIYA64CXVDRYSYYDANCNFSM4OJK3DFQ>
> .
>

JerrySievert commented:
> Anything created after 8.2 would never have OPAQUEOID as a return type

that's about what I thought.  I went and checked my notes, and supposedly we're still able to support all the way down to 8.1 (I haven't tried to compile against it in a while).

let me give this some thought, as to whether to deprecate anything older than, say, 9.1 for this version.

jim-mlodgenski commented:
On Fri, Jun 26, 2020 at 11:09 AM Jerry Sievert <notifications@github.com>
wrote:

> Anything created after 8.2 would never have OPAQUEOID as a return type
>
> that's about what I thought. I went and checked my notes, and supposedly
> we're still able to support all the way down to 8.1 (I haven't tried to
> compile against it in a while).
>
> let me give this some thought, as to whether to deprecate anything older
> than, say, 9.1 for this version.
>
The main PG community only supports 9.5 on up at this point with 9.5 being
deprecated in 8 months. Anyone still running 9.1 or older has been running
on an unsupported version for a very long time.


> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plv8/plv8/pull/397#issuecomment-650231497>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJFFZZVEBCM3HOK5FGXKEDRYS24LANCNFSM4OJK3DFQ>
> .
>

JerrySievert commented:
> The main PG community only supports 9.5 on up at this point with 9.5 being
deprecated in 8 months. Anyone still running 9.1 or older has been running
on an unsupported version for a very long time.

yes, I realize that, but plv8 has a different deprecation schedule, because of a set of users.

not to mention that I still see posts on pg-bugs on a fairly regular basis with people still stuck using older versions of pg.

those unsupported versions can still be very hard to move off of, especially in a larger group that is risk adverse.

jim-mlodgenski commented:
On Fri, Jun 26, 2020 at 11:20 AM Jerry Sievert <notifications@github.com>
wrote:

> The main PG community only supports 9.5 on up at this point with 9.5 being
> deprecated in 8 months. Anyone still running 9.1 or older has been running
> on an unsupported version for a very long time.
>
> yes, I realize that, but plv8 has a different deprecation schedule,
> because of a set of users.
>
> not to mention that I still see posts on pg-bugs on a fairly regular basis
> with people still stuck using older versions of pg.
>
> those unsupported versions can still be very hard to move off of,
> especially in a larger group that is risk adverse.
>

To be conservative then, we can just put an "if PG_VERSION_NUM >= 130000"
around the OPAQUEOID change and things would continue to work the same way
on older versions.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plv8/plv8/pull/397#issuecomment-650237131>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJFFZ2K5KI3WH3T3YDCWW3RYS4D5ANCNFSM4OJK3DFQ>
> .
>

JerrySievert commented:
that would be a fantastic compromise.  would you be willing to make that change on your PR?

jim-mlodgenski committed:
Changes to build against PG13. This includes the removal of a reference to OPAQUEOID which has been removed.

jim-mlodgenski committed:
Update plv8.cc

Add an check for the PostgreSQL version around the OPAQUEOID change to keep the behavior the same in older versions of PostgreSQL.

jim-mlodgenski commented:
On Fri, Jun 26, 2020 at 11:24 AM Jerry Sievert <notifications@github.com>
wrote:

> that would be a fantastic compromise. would you be willing to make that
> change on your PR?
>

No problem. I made the change.

—
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plv8/plv8/pull/397#issuecomment-650239407>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJFFZ2H67LN3X74MOV5TA3RYS4TJANCNFSM4OJK3DFQ>
> .
>

JerrySievert commented:
awesome, thanks very much for the PR!  I'll let the CI run and will plan to do a merge this afternoon (pacific time).

I can do the back port to the 2.x branch.
